custom_components/tesla_custom/button.py

```diff
@@ -1,84 +1,147 @@
 
-class FlashLights(TeslaDevice, ButtonEntity):
-    """Representation of a Tesla flash lights button."""
 
-    def __init__(self, tesla_device, coordinator):
-        """Initialise the button."""
-        super().__init__(tesla_device, coordinator)
-        self.controller = coordinator.controller
+class Horn(TeslaCarDevice, ButtonEntity):
+    """Representation of the Tesla Battery Sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        car: TeslaCar,
+        coordinator: TeslaDataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Sensor Entity."""
+        super().__init__(hass, car, coordinator)
+        self.type = "horn"
+        self._attr_icon = "mdi:bullhorn"
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self._car.honk_horn()
+
+
+class FlashLights(TeslaCarDevice, ButtonEntity):
+    """Representation of the Tesla Battery Sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        car: TeslaCar,
+        coordinator: TeslaDataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Sensor Entity."""
+        super().__init__(hass, car, coordinator)
+        self.type = "flash lights"
+        self._attr_icon = "mdi:car-light-high"
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self._car.flash_lights()
+
+
+class WakeUp(TeslaCarDevice, ButtonEntity):
+    """Representation of the Tesla Battery Sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        car: TeslaCar,
+        coordinator: TeslaDataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Sensor Entity."""
+        super().__init__(hass, car, coordinator)
+        self.type = "wake up"
+        self._attr_icon = "mdi:moon-waning-crescent"
+        self._attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self._car.wake_up()
+
+
+class ForceDataUpdate(TeslaCarDevice, ButtonEntity):
+    """Representation of the Tesla Battery Sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        car: TeslaCar,
+        coordinator: TeslaDataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Sensor Entity."""
+        super().__init__(hass, car, coordinator)
+        self.type = "force Data update"
+        self._attr_icon = "mdi:database-sync"
+        self._attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self.update_controller(wake_if_asleep=True, force=True)
 
-    @TeslaDevice.Decorators.check_for_reauth
-    async def async_press(self, **kwargs):
-        """Send the command."""
-        _LOGGER.debug("Flash lights: %s", self.name)
-        await self.tesla_device.flash_lights()
+    @property
+    def available(self) -> bool:
```

Comment:
What's the purpose of defining this method for only this entity?

Reply:
I believe it's to override the parent class which will set `False` if the vehicle data is unavailable (empty dict). Since this button is to force a data update, it makes sense to always keep it enabled so you can try to force a data update.